lua_modules/sntp/sntp.lua

```diff
@@ -0,0 +1,290 @@
+local MAX_SERVER_ATTEMPTS = 2
+local SNTP_TIMEOUT        = 5000
+
+-- Constructor
+--   sc and fc are our Success and Failure Callbacks, resp.
+local new = function(serv, sc, fc, now)
+
+  if     type(serv) == "string" then serv = {serv}
+  elseif serv == nil            then serv =
+    {
+      nil,
+      "1.nodemcu.pool.ntp.org",
+      "2.nodemcu.pool.ntp.org",
+    }
+    local ni = net.ifinfo(0)
+    ni = ni and ni.dhcp
+    serv[1] = ni.ntp_server or "0.nodemcu.pool.ntp.org"
+  elseif type(serv) ~= "table"  then error "Bad server table"
+  end
+
+  if type(sc) ~= "function" and type(sc) ~= "lightfunction" then
```

Comment:
The type "lightfunction" no longer exists, so all of these clauses are redundant.  All functions are type "function". (This is for compatibility with Lua53)